src/device/protocol/buttplugIo/buttplugIoDeviceFactory.ts

```diff
@@ -21,11 +21,22 @@ export default class ButtplugIoDeviceFactory
 
     private readonly logger: Logger;
 
+    private useNameAsSerial: boolean
+
     public constructor(uuidFactory: UuidFactory, dateFactory: DateFactory, settings: Settings, logger: Logger) {
         this.uuidFactory = uuidFactory;
         this.dateFactory = dateFactory;
         this.settings = settings;
         this.logger = logger;
+
+        this.useNameAsSerial = false;
+        const configuredDeviceSources = this.settings.getDeviceSources();
+        configuredDeviceSources.forEach((value) => {
+            if (value.type === 'buttplugIoWebsocket' && value.config?.useNameAsSerial) {
+                this.logger.debug(`Device source ${value.type} using names as serial`);
+                this.useNameAsSerial = true;
+            }
+        });
```

Comment:
This should be done in the `buttplugIoWebsocketDeviceProviderFactory` class, there you get the config for the device source. See:

https://github.com/SlvCtrlPlus/slvctrlplus-server/blob/main/src/device/provider/deviceProviderLoader.ts#L41

Reply:
In fact, I did it there but I don't use it in the deviceProviderFactory.

But when constructing the device factory (here https://github.com/drone4770/slvctrlplus-server/blob/9a3b13b5932dfc9cf48ab2b65b5026adfa370f24/src/serviceProvider/deviceServiceProvider.ts#L82), we don't get the config of the source provider, only the full settings.

So two choices: 

1. I pass the value as a parameter at each device creation to the create() method of the deviceFactory and then to the createKnownDevice() method where I use it (here https://github.com/SlvCtrlPlus/slvctrlplus-server/blob/771a3c9a66e10d2f73cd36a29abdd173e692e13f/src/device/protocol/buttplugIo/buttplugIoDeviceFactory.ts#L31 and here https://github.com/SlvCtrlPlus/slvctrlplus-server/blob/771a3c9a66e10d2f73cd36a29abdd173e692e13f/src/device/protocol/buttplugIo/buttplugIoDeviceFactory.ts#L86)
2. I do it like that once with an ugly loop when the class is instantiated at startup like I did

I don't really like the second choice, but didn't really like the first one either.

Reply:
I see. Maybe we should have a DeviceFactory per source. Currently all sources of the same type share the same device factory.

If every source has it's own device factory you can configure the device factory properly for each source... would this make things prettier? Wdyt?

Reply:
I quickly played it through. I think this would be nicer -> https://github.com/SlvCtrlPlus/slvctrlplus-server/pull/28/files

Reply:
I agree it can be a bit overwhelming first. But with your solution you force all buttplugIo sources to use either device name or id, but you can configure it per source. So the code doesn’t really behave correctly. I rather have consistent code which is a bit more complex than code that behaves „inconsistent“.

If there‘s an easier way or a better naming for the Factory of Factories I‘m happy to add it. But I think the suggestion I made is the most „consistent“ one.

Reply:
I didn't really see you were planning to maybe have another source using buttplugIoDevice not using the websocket.
Then we should use the option 1 in my previous comment and pass it from the device source instead of adding another layer of complication.

Reply:
The idea is that you can connect multiple buttplug.io servers at the same time. Maybe a frined stops by with the intiface central. Then you could just hook into their Intiface Central without reconnecting all the devices they bring to your intiface first. I just realized that we should also make the device source uuid part of the device id as well. But I can patch this after we got your PR into main.

Indeed passing it from the device provider over the device factory `create` method is enough. Forget about my factory of factories. ;-) Although I want to remove probably in a later iteration the whole "known device" stuff from the device factory and move it into the provider. But that can be done easily at a later point.

Reply:
Ok , when you will do that you I think you will have to change other things, because I saw the source name is hardcoded in buttplugIoWebSocketDeviceProfider.ts